### ISSUE TITLE
Comparison between NonZeroUsizes doesn't inhibit const suggestion.

### ISSUE THREAD

**Issue by SUPERCILEX** [rust-lang/rust-clippy#9646]
### Lint Name

missing-const-for-fn

### Reproducer

I tried this code:

```rust
    pub fn new(
        num_files: NonZeroUsize,
        file_to_dir_ratio: NonZeroUsize,
    ) -> std::result::Result<Self, NumFilesWithRatioError> {
        if file_to_dir_ratio > num_files {
            return Err(NumFilesWithRatioError::InvalidRatio {
                num_files,
                file_to_dir_ratio,
            });
        }

        Ok(Self {
            num_files,
            file_to_dir_ratio,
        })
    }
```

I saw this happen after making that function const:

```
error[E0277]: can't compare `std::num::NonZeroUsize` with `std::num::NonZeroUsize` in const contexts
  --> src/generator.rs:65:30
   |
65 |         if file_to_dir_ratio > num_files {
   |                              ^ no implementation for `std::num::NonZeroUsize < std::num::NonZeroUsize` and `std::num::NonZeroUsize > std::num::NonZeroUsize`
   |
   = help: the trait `~const std::cmp::PartialOrd` is not implemented for `std::num::NonZeroUsize`
```

I expected to see this happen:

### Version

```text
rustc 1.66.0-nightly (6b3ede3f7 2022-10-13)
binary: rustc
commit-hash: 6b3ede3f7bc502eba7bbd202b4b9312d812adcd7
commit-date: 2022-10-13
host: x86_64-unknown-linux-gnu
release: 1.66.0-nightly
LLVM version: 15.0.2
```

**Comment by SUPERCILEX** [comment]
Seems to be fixed.
